python/tidy/servo_tidy/tidy.py

```diff
@@ -302,6 +307,44 @@ def check_toml(file_name, lines):
             yield (idx + 1, "found asterisk instead of minimum version number")
 
 
+def check_shell(file_name, lines):
+    if not file_name.endswith(".sh"):
+        raise StopIteration
+
+    shebang = "#!/usr/bin/env bash"
+    required_options = {"set -o errexit", "set -o nounset", "set -o pipefail"}
+    # A flag to indicate whether we've found a "real" statement in the script, e.g. not one of `set -o <option>`
+    found_script_statement = False
+
+    for idx, line in enumerate(lines):
+        if idx == 0:
+            if not line.startswith(shebang):
+                yield (idx + 1, 'script does not start with "{}"'.format(shebang))
+        else:
+            if found_script_statement:
+                break
+            else:
+                stripped = line.rstrip()
+
+                # Comments or blank lines are ignored. (Trailing whitespace is caught with a separate linter.)
+                if line.startswith("#") or stripped == "":
+                    continue
+                elif stripped in required_options:
+                    required_options.remove(stripped)
+                else:
+                    # The first non-comment, non-whitespace, non-option line is the first "real" line of the script.
+                    # The shebang, options, etc. must come before this.
+                    found_script_statement = True
```

Comment:
I think you can get rid of `found_script_statement` and just `break` after the `if len(required_options) != 0` block, instead of doing it on the next iteration.